Gooey/Sources/Color.swift

```diff
@@ -0,0 +1,217 @@
+import Foundation
+
+/// Provides a unified convenience API for colors with a focus on brevity.
+public struct Color {
+
+    /// The location of the alpha component in representations of this value.
+    public enum AlphaLocation {
+        
+        /// The alpha component comes before the color components.
+        case leading
+        
+        /// The alpha component comes after the color components.
+        case trailing
+        
+    }
+    
+    /// The red component of this color.
+    public var red: UInt8
+    
+    /// The green component of this color.
+    public var green: UInt8
+    
+    /// The blue component of this color.
+    public var blue: UInt8
+    
+    /// The alpha component of this color.
+    public var alpha: UInt8
+    
+    /// Returns the red component of this color represented as a percent.
+    public var percentRed: CGFloat {
+        return red.percent
+    }
+    
+    /// Returns the green component of this color represented as a percent.
+    public var percentGreen: CGFloat {
+        return green.percent
+    }
+    
+    /// Returns the blue component of this color represented as a percent.
+    public var percentBlue: CGFloat {
+        return blue.percent
+    }
+    
+    /// Returns the alpha component of this color represented as a percent.
+    public var percentAlpha: CGFloat {
+        return alpha.percent
+    }
+    
+    /// Returns the intensity or saturation of the color.
+    public var intensity: UInt8 {
+        let redFactor = 0.299 * Double(red)
+        let greenFactor = 0.587 * Double(green)
+        let blueFactor = 0.114 * Double(blue)
+        return UInt8(redFactor + greenFactor + blueFactor)
+    }
+    
+    /// Returns whether light colors will appear natural alongside this color.
+    ///
+    /// This is most useful for determining whether to select a light or dark text color for text
+    /// super-imposed over this color.
+    public var complementsLightColors: Bool {
+        return intensity < 186
+    }
+    
+    /// Returns the `UIColor` which corresponds with this color.
+    public var uiColor: UIColor {
```

Comment:
Now that I think about it, is it because `Gooey.h` is a public header which imports UIKit so all of UIKit is dumped in every Swift file? Pure speculation.

Reply:
I buy it